lib/itertools.py

```diff
@@ -17,15 +17,16 @@
 import _collections
 import sys
 
+def _from_iterable(cls, iterable):
```

Comment:
This two lines supposed to be inside chain class, I think. 
Something like 
https://docs.python.org/2/whatsnew/2.4.html#pep-318-decorators-for-functions-and-methods

Any issue with inside the class?